src/toxcore/dht_new/packet.rs

```diff
@@ -0,0 +1,768 @@
+/*
+    Copyright (C) 2013 Tox project All Rights Reserved.
+    Copyright © 2016-2017 Zetok Zalbavar <zexavexxe@gmail.com>
+    Copyright © 2018 Namsoo CHO <nscho66@gmail.com>
+
+    This file is part of Tox.
+
+    Tox is libre software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Tox is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Tox.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+// ↓ FIXME expand doc
+/*! DHT part of the toxcore.
+
+    * takes care of the serializing and de-serializing DHT packets
+    * ..
+*/
+
+
+// use byteorder::{
+//     BigEndian,
+//     LittleEndian,
+//     NativeEndian,
+//     WriteBytesExt
+// };
+use nom::{le_u8, le_u16, be_u64, be_u16, rest};
+
+//use std::cmp::{Ord, Ordering};
+//use std::convert::From;
+//use std::fmt::Debug;
+use std::net::{
+    IpAddr,
+    Ipv4Addr,
+    Ipv6Addr,
+    SocketAddr,
+    SocketAddrV4,
+    SocketAddrV6
+};
+//use std::ops::Deref;
+
+use toxcore::dht_new::binary_io::*;
+use toxcore::crypto_core::*;
+//use toxcore::packet_kind::PacketKind;
+
+use toxcore::dht_new::cookie_factory::*;
+
+/** Standard DHT packet that encapsulates in the encrypted payload
```

Comment:
Actually it is not encrypted. Don't use PacketKind in docs since we will not use the struct itself.

Better point to docs: https://zetok.github.io/tox-spec/#dht